apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax.Grpc;
+using Google.Cloud.Spanner.Common.V1;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using static Google.Cloud.Spanner.V1.PoolRewrite.SessionPool;
+
+namespace Google.Cloud.Spanner.V1.PoolRewrite.Tests
+{
+    public partial class SessionPoolTests
+    {
+        /// <summary>
+        /// Tests for <see cref="SessionPool.TargetedSessionPool" />, mostly
+        /// involving direct construction of the pool to avoid any maintenance loops etc.
+        /// </summary>
+        public sealed class TargetedSessionPoolTests
+        {
+            private static readonly DatabaseName s_databaseName = new DatabaseName("project", "instance", "database");
+
+            private static TargetedSessionPool CreatePool(bool acquireSessionsImmediately)
+            {
+                var client = new SessionTestingSpannerClient();
+                // Fixed session pool options so we can hard-code values without worrying about the defaults changing
+                var options = new SessionPoolOptions
+                {
+                     IdleSessionRefreshDelay = TimeSpan.FromMinutes(15),
+                     MaintenanceLoopDelay = TimeSpan.FromMinutes(1),
+                     PoolEvictionDelay = TimeSpan.FromMinutes(100), // Deliberately not a multiple of 15
+                     SessionEvictionJitter = RetrySettings.NoJitter,
+                     SessionRefreshJitter = RetrySettings.NoJitter,
+                     MaximumActiveSessions = 100,
+                     MaximumConcurrentSessionCreates = 20,
```

Comment:
Now that I see this option, I don't remebmer seing it used anywhere, in particular it should be used in `StartAcquisitionTasksIfNecessary` no?

Reply:
Yes, I remember now from the big PR.